core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+/// The function generates URL for any member function (including Constructor/
+/// Destructor) of "className". Doxygen uses specific syntax - the first part
+/// of the URL is only for the class and it is generated with the help of
+/// "Url". Then we append "#a" and sequence of numbers and letters.
+/// This sequence is text, encrypted with MD5. The original text is
+/// "Return type className::methodNamemethodName(arguments)".
+/// For constructor/destructor of the class the return type is not appended.
+///
+/// \param[in] className <the name of the class
+/// \param[in] methodName <the name of the method from the class
+/// \param[in] type <enumerator for method or constructor
+
+static TString URLforClassMember(const TString &className, TClass *classVariable,  const TString &methodName, EClassURL type)
+{
+   // We generate the URL for the class.
+   TString url(UrlGenerator(className, kURLforClass));
+   url.Append("#a");
+   TString md5Text("");
+   // The use of enumeration is not necessary, but it will allows for easy
+   // future improvements.
+   if (type == kURLforConstructor) {
+      // In the case of constructor/destructor the "Return type" is missing.
+      md5Text.Append(className);
+   } else if (type == kURLforMethod) {
+       // In the case of method, we append the return type too.
+       // "FormatReturnTypeForDoxygen" modifies the return type with respect to Doxygen's reqirement.
+       md5Text.Append((FormatReturnTypeForDoxygen(className, classVariable,  methodName)));
+       md5Text.Append(" ");
+       md5Text.Append(className);
+   }
+   md5Text.Append("::");
+   md5Text.Append(methodName);
+   md5Text.Append(methodName);
+   // We use "FormatMethodForDoxygen" to modify the arguments of Method with respect of Doxygen.
+   md5Text.Append(FormatMethodForDoxygen(className, classVariable,  methodName));
+   url.Append(md5Text.MD5());
+   return url;
+}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// The function lists useful commands (".help") or opens the online reference
+/// guide, generated with Doxygen, for a certain class (".help className"),
+/// method of a class (".help className::Method") or a namespace
+/// (".help NameSpace").
+///
+/// \param[in] line <command from the command line
+
+void TApplication::Help(const char *line)
+{
+   // If the user types ".help className::Method" in the command line we need
+   // to extract the className and Method. We separate the input text
+   // into two parts (before and after the "::" sign) and save them in
+   // "input" - a vector of chars.
+   std::vector<char *> input;
+   char *in_token = strtok((char *)line, " ");
+   while (in_token != nullptr) {
+      in_token = strtok(nullptr, "::");
+      input.push_back(in_token);
+   }
+   // "input[0]" is the class name. We first check if the user wants to use the
+   // extended ".help" command or the old one.
```

Comment:
The check "we have an argument to .help" should happen independently of "do we have a scope", and before any tokenization operation. First thing in the function. E.g.:
- remove all training space
- does the line consist only of ".help"? command help, else reference doc.